vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -35,16 +40,29 @@
     @Autowired
     private SequenceRepository sequenceRepo;
     
+    
+    @Autowired
+    private BranchingPointRepository bpointRepo;
+    
+    @Autowired
+    private ChoiceRepository choiceRepo;
+    
     private final Logger logger = LoggerFactory.getLogger(getClass());
     
-    @Override
-    public ISlide createSlide(String moduleId, SlideForm slideForm) {
-        IModule module = moduleManager.getModule(moduleId);
-        ISlide slide = slideFactory.createSlide(module, slideForm);
-        slideRepo.save((Slide) slide);
-        return slide;
-    }
-
+    @Override
+    public ISlide createSlide(IModule module, SlideForm slideForm, SlideType type) {
+        ISlide slide = slideFactory.createSlide(module, slideForm, type);             
+        slideRepo.save((Slide) slide);        
+        return slide;
+    }
+
+    @Override
+    public IBranchingPoint createBranchingPoint(IModule module, SlideForm slideForm, SlideType type) {
+        ISlide branchingPoint = slideFactory.createSlide(module, slideForm, type);            
+        bpointRepo.save((BranchingPoint) branchingPoint);        
```

Comment:
you want to return the branching point that the save method returns (it'll have the id)